src/components/interaction-designer/Block.vue

```diff
@@ -131,87 +132,80 @@
           @mouseenter="isConnectionSourceRelocateActive && activateExitAsDropZone($event, exit)"
           @mouseleave="isConnectionSourceRelocateActive && deactivateExitAsDropZone($event, exit)">
           <div v-if="!(exit.default && block.vendor_metadata.io_viamo.noValidResponse === NoValidResponseHandler.END_CALL)">
-          <div class="total-label-container">
-            <span class="badge badge-primary tree-block-item-label tree-block-item-output-subscribers-1" />
-          </div>
+            <div class="total-label-container">
+              <span class="badge badge-primary tree-block-item-label tree-block-item-output-subscribers-1" />
+            </div>
 
-          <div
-            class="block-exit-tag badge badge-warning"
-            @mouseenter="exitMouseEnter(exit)"
-            @mouseleave="exitMouseLeave(exit)">
-            <span class="block-exit-tag-text align-self-center"
-                  v-b-tooltip.hover.top="exit.test">
-              {{exit.name || '(untitled)'}}
-            </span>
-
-            <span
-              class="align-self-center">
-              <template v-if="exit.destination_block == null">
-                <plain-draggable
-                  :id="`exit/${exit.uuid}/pseudo-block-handle`"
-                  :key="`exit/${exit.uuid}/pseudo-block-handle`"
-                  v-b-tooltip.hover.top="transIf(isEditable, 'flow-builder.tooltip-new-connection')"
-                  class="btn btn-xs"
-                  :is-editable="isEditable"
-                  @initialized="handleDraggableInitializedFor(exit, $event)"
-                  @dragStarted="onCreateExitDragStarted($event, exit)"
-                  @dragged="onCreateExitDragged($event)"
-                  @dragEnded="onCreateExitDragEnded($event, exit)"
-                  @destroyed="handleDraggableDestroyedFor(exit)">
-                  <i
-                    v-if="isEditable && exitHovers[exit.uuid]"
-                    class="glyphicon glyphicon-move" />
-                  <font-awesome-icon
-                    v-else
-                    class="text-danger"
-                    :icon="['far', 'dot-circle']" />
-                </plain-draggable>
-
-                <template v-if="isConnectionCreateActive && isExitActivatedForCreate(exit) && livePosition">
+            <div
+              class="block-exit-tag badge badge-warning d-flex justify-content-center"
+              :style="{background: exitBackgroundColor(exit)}"
```

Comment:
Personally, I think these templates are clearest if we can take the approach of applying the semantic "state" of things using CSS classes, then implement the details of styling in the CSS. This gives us very clean [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns), keeping CSS + HTML out of JS and JS out of template!

An example (pls bear with the pseudo-code 😛 ):
```
<template>
...
    <div
      v-for="(exit, key) in block.exits"
      :key="exit.uuid"
      class="block-exit"
      :class="{
        ['is-hovered']: isExitHovered(exit),
        ['is-not-hovered']: !isExitHovered(exit),
        ['is-connected']: doesExitHaveConnection(exit),
        ['is-disconnected']: !doesExitHaveConnection(exit),
      <div 
        class="block-exit-tag ..."
        ... >
          {{ exit.name || '(untitled)' }}
         ...
</template>
```

```
<script>
...
methods: {
  isExitHovered(exit) {
    return exit.destination_block != null && (this.exitHovers[exit.uuid] || this.lineHovers[exit.uuid])
  },

  doesExitHaveConnection(exit) {
    return exit.destination_block != null
  }
  ...
}
</script>
```

```
<style>
.block {
  .block-exit {
    // default styling
    background: #xxx;

    // state cascaded styling
    &.is-connected {
      background: #xxx;
    }

    &.is-disconnected {
      background: #xxx;      
    }

    &.is-hovered {
      background: #xxx;      
    }
  }
}
<style>
```

---

Sorry for the verbosity— what do you think of this approach?

Reply:
I like this approach!  This solved the issue of text contrast with the button too 😆  Thanks